MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,275 @@
+- [Core-js](https://www.npmjs.com/package/core-js): Used to polyfill a couple of things, also mostly on IE11, Opera Mini, and some mobile-specific browsers:
+  - [string/startsWith](https://caniuse.com/?search=startsWith)
+  - [string/includes](https://caniuse.com/es6-string-includes)
+  - [set](https://caniuse.com/mdn-javascript_builtins_set)
+  - [symbol](https://caniuse.com/mdn-javascript_builtins_symbol)
+  - [array/from](https://caniuse.com/mdn-javascript_builtins_array_from)
+  - [array/includes](https://caniuse.com/array-includes)
+- [fast-text-encoding](https://www.npmjs.com/package/fast-text-encoding): Used to polyfill TextEncoder and TextDecoder on IE11 and Opera Mini.
+- [unfetch](https://www.npmjs.com/package/unfetch): Used to [ponyfill fetch on IE11](https://caniuse.com/?search=fetch).
+
+By removing these polyfills, the bundle size for Auth0-SPA-JS has dropped 60%. As this is a core dependency of Auth0-Vue this ensures your users have a better experience when integrating Auth0 into your application using Auth0-Vue.
+
+## Public API Changes
+
+With the release of this new major version, some changes were made that impact the public API of Auth0-Vue. If you are using TypeScript, these should be flagged for you. However we still recommend reviewing this list thoroughly as some changes are changes in behavior.
+
+### Introduction of `authorizationParams`
+
+A breaking change that will affect pretty much everyone is the introduction of `authorizationParams`, a more structured approach to providing parameters - including custom parameters - to Auth0.
+
+In v1, objects passed to our methods are always a mix of properties used for configuring the SDK and properties with the sole purpose to pass through to Auth0.
+
+```ts
+app.use(
+  createAuth0({
+    domain: '<AUTH0_DOMAIN>',
+    clientId: '<AUTH0_CLIENT_ID>',
+    redirect_uri: '<MY_CALLBACK_URL>'
+  })
+);
+```
+
+With v2 of our SDK, we have improved the API by separating those properties used to configure the SDK, from properties that are sent to Auth0. The SDK configuration properties will stay on the root, while any property that should be sent to Auth0 should be set on `authorizationParams`.
+
+```ts
+app.use(
+  createAuth0({
+    domain: '<AUTH0_DOMAIN>',
+    clientId: '<AUTH0_CLIENT_ID>',
+    authorizationParams: {
+      redirect_uri: '<MY_CALLBACK_URL>'
+    }
+  })
+);
+```
+
+The above changes affect the following methods:
+
+- loginWithRedirect
+- loginWithPopup
+- getAccessTokenWithPopup
+- getAccessTokenSilently
+
+### Introduction of `logoutParams`
+
+In v1, `logout` can be called with an object containing a number of properties, both a mix between properties used to configure the SDK as well as those used to pass through to Auth0.
+
+With v2, logout now takes an object that can only contain two properties, `clientId` and `logoutParams`.
```

Comment:
It will also contain `onRedirect`/`openUrl`. Adding a comment to ensure we don't forget to update.